File: core/src/galenscovell/hinterstar/Hinterstar.scala
package galenscovell.hinterstar

import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.{Game, Gdx}
import galenscovell.hinterstar.ui.screens._
import galenscovell.hinterstar.util._


class Hinterstar extends Game {
  var spriteBatch: SpriteBatch = _
  var loadScreen: LoadScreen = _
  var mainMenuScreen: MainMenuScreen = _
  var startScreen: StartScreen = _
  var gameScreen: GameScreen = _


  def create(): Unit =  {
    PlayerData.init()
    spriteBatch = new SpriteBatch
    loadScreen = new LoadScreen(this)
    mainMenuScreen = new MainMenuScreen(this)
    setScreen(loadScreen)
  }

  override def dispose(): Unit =  {
    loadScreen.dispose()
    mainMenuScreen.dispose()
    if (startScreen != null) {
      startScreen.dispose()
    }
    if (gameScreen != null) {
      gameScreen.dispose()
    }
    Resources.dispose()
  }

  def createStartScreen(): Unit = {
    if (startScreen != null) {
      startScreen.dispose()
    }
    startScreen = new StartScreen(this)
  }

  def createGameScreen(): Unit =  {
    if (gameScreen != null) {
      gameScreen.dispose()
    }
    gameScreen = new GameScreen(this)
  }

  def loadGame(): Unit = {
    // TODO: Not yet implemented, PlayerData.load everything

  }

  def createPreferenceScreen(): Unit =  {
    // TODO: Not yet implemented

  }

  def quitGame(): Unit = {
    Gdx.app.exit()
  }
}

File: core/src/galenscovell/hinterstar/Hinterstar.kt
package galenscovell.hinterstar

import com.badlogic.gdx.*
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import galenscovell.hinterstar.ui.screens.*
import galenscovell.hinterstar.util.*


class Hinterstar() : Game() {
    lateinit var spriteBatch: SpriteBatch
    lateinit var loadScreen: LoadScreen
    lateinit var mainMenuScreen: MainMenuScreen
    var startScreen: StartScreen? = null
    var gameScreen: GameScreen? = null



    override fun create(): Unit {
        PlayerData.init()
        spriteBatch = SpriteBatch()
        loadScreen = LoadScreen(this)
        mainMenuScreen = MainMenuScreen(this)
        setScreen(loadScreen)
    }

    override fun dispose(): Unit {
        loadScreen.dispose()
        mainMenuScreen.dispose()
        startScreen?.dispose()
        gameScreen?.dispose()
        Resources.dispose()
    }

    fun createStartScreen(): Unit {
        startScreen?.dispose()
        startScreen = StartScreen(this)
    }

    fun createGameScreen(): Unit {
        gameScreen?.dispose()
        gameScreen = GameScreen(this)
    }

    fun loadGame(): Unit {
        // TODO: Not yet implemented, PlayerData.load() everything
    }

    fun createPreferenceScreen(): Unit {
        // TODO: Not yet implemented
    }

    fun quitGame(): Unit {
        Gdx.app.exit()
    }
}
